truffle/src/main/java/org/jruby/truffle/core/kernel/KernelNodes.java

```diff
@@ -710,12 +714,21 @@ public Object exec(VirtualFrame frame, Object command, Object[] args) {
 
         @TruffleBoundary
         private String[] buildCommandLine(Object command, Object[] args) {
-            final String[] commandLine = new String[1 + args.length];
-            commandLine[0] = command.toString();
+            final List<String> commandLine = new ArrayList<>(1 + args.length);
+            if (RubyGuards.isRubyArray(command)) {
+                final Object[] store = (Object[]) getStore((DynamicObject) command);
+                commandLine.add(store[0].toString());
```

Comment:
I see, please mention the `exec([cmdname, argv0], arg1, ...)` there to clarify the meaning.